Build failure: suitesparse

## Steps To Reproduce

```sh
NIX_PATH=nixpkgs=https://github.com/NixOS/nixpkgs/archive/88195a94f390381c6afcdaa933c2f6ff93959cb4.tar.gz
nix-build --no-out-link --expr 'let pkgs = import <nixpkgs> { config = { allowUnfree = true; cudaSupport = true; }; }; in pkgs.suitesparse'
```

## Build log

<details>

<summary>Build Log</summary>

```
Sourcing setup-cuda-hook
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/hl097s04bc6v3yr18isiicq444lgcr6l-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Executing setupCUDAToolkitCompilers
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash INSTALL=/nix/store/hbr3ma1fzvwr7zvn4sp67wgwmayals9x-suitesparse-5.13.0 INSTALL_INCLUDE=/nix/store/lyv9aigvn1crwlmrr7334902b92izvak-suitesparse-5.13.0-dev/include JOBS=\$\(NIX_BUILD_CORES\) MY_METIS_LIB=-lmetis CUDA_PATH=/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99 CUDART_LIB=/nix/store/b5arfaz3xqrspi891903m9p66q4vhl5p-cuda_cudart-12.4.99-lib/lib/libcudart.so CUBLAS_LIB=/nix/store/5x4cw7hfnky0cb4jhkrzv4gk25m82dg4-libcublas-12.4.2.65-lib/lib/libcublas.so library
Using pre-installed METIS 5.1.0 library at  [-lmetis]
( cd SuiteSparse_config && make )
make[1]: Entering directory '/build/source/SuiteSparse_config'
gcc    -O3 -fexceptions -fPIC -fopenmp -c SuiteSparse_config.c
ar -rv libsuitesparseconfig.a SuiteSparse_config.o
ar: creating libsuitesparseconfig.a
a - SuiteSparse_config.o
ranlib libsuitesparseconfig.a
make install INSTALL=/build/source
make[2]: Entering directory '/build/source/SuiteSparse_config'
gcc -L/build/source/lib -Wl,-rpath=/build/source/lib -shared -Wl,-soname -Wl,libsuitesparseconfig.so.5 -Wl,--no-undefined -Wl,-rpath,'$ORIGIN' -Wl,-z,origin SuiteSparse_config.o -o /build/source/lib/libsuitesparseconfig.so.5.13.0 -lm -lrt
( cd /build/source/lib ; ln -sf libsuitesparseconfig.so.5.13.0 libsuitesparseconfig.so )
( cd /build/source/lib ; ln -sf libsuitesparseconfig.so.5.13.0 libsuitesparseconfig.so.5 )
cp -f SuiteSparse_config.h /nix/store/lyv9aigvn1crwlmrr7334902b92izvak-suitesparse-5.13.0-dev/include
cp -f README.txt /build/source/share/doc/suitesparse-5.13.0/SUITESPARSECONFIG_README.txt
chmod 755 /build/source/lib/libsuitesparseconfig.so.5.13.0
chmod 755 /build/source/lib/libsuitesparseconfig.so
chmod 644 /nix/store/lyv9aigvn1crwlmrr7334902b92izvak-suitesparse-5.13.0-dev/include/SuiteSparse_config.h
chmod 644 /build/source/share/doc/suitesparse-5.13.0/SUITESPARSECONFIG_README.txt
make[2]: Leaving directory '/build/source/SuiteSparse_config'
make[1]: Leaving directory '/build/source/SuiteSparse_config'
( cd AMD && make library )
make[1]: Entering directory '/build/source/AMD'
( cd Lib    ; make )
make[2]: Entering directory '/build/source/AMD/Lib'
make install INSTALL=/build/source
make[3]: Entering directory '/build/source/AMD/Lib'
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/amd_aat.c -o amd_i_aat.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/amd_1.c -o amd_i_1.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/amd_2.c -o amd_i_2.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/amd_dump.c -o amd_i_dump.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/amd_postorder.c -o amd_i_postorder.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/amd_defaults.c -o amd_i_defaults.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/amd_post_tree.c -o amd_i_post_tree.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/amd_order.c -o amd_i_order.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/amd_control.c -o amd_i_control.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/amd_info.c -o amd_i_info.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/amd_valid.c -o amd_i_valid.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/amd_preprocess.c -o amd_i_preprocess.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/amd_aat.c -o amd_l_aat.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/amd_1.c -o amd_l_1.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/amd_2.c -o amd_l_2.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/amd_dump.c -o amd_l_dump.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/amd_postorder.c -o amd_l_postorder.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/amd_defaults.c -o amd_l_defaults.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/amd_post_tree.c -o amd_l_post_tree.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/amd_order.c -o amd_l_order.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/amd_control.c -o amd_l_control.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/amd_info.c -o amd_l_info.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/amd_valid.c -o amd_l_valid.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/amd_preprocess.c -o amd_l_preprocess.o
ar -rv libamd.a amd_i_aat.o amd_i_1.o amd_i_2.o amd_i_dump.o amd_i_postorder.o amd_i_defaults.o amd_i_post_tree.o amd_i_order.o amd_i_control.o amd_i_info.o amd_i_valid.o amd_i_preprocess.o amd_l_aat.o amd_l_1.o amd_l_2.o amd_l_dump.o amd_l_postorder.o amd_l_defaults.o amd_l_post_tree.o amd_l_order.o amd_l_control.o amd_l_info.o amd_l_valid.o amd_l_preprocess.o
ar: creating libamd.a
a - amd_i_aat.o
a - amd_i_1.o
a - amd_i_2.o
a - amd_i_dump.o
a - amd_i_postorder.o
a - amd_i_defaults.o
a - amd_i_post_tree.o
a - amd_i_order.o
a - amd_i_control.o
a - amd_i_info.o
a - amd_i_valid.o
a - amd_i_preprocess.o
a - amd_l_aat.o
a - amd_l_1.o
a - amd_l_2.o
a - amd_l_dump.o
a - amd_l_postorder.o
a - amd_l_defaults.o
a - amd_l_post_tree.o
a - amd_l_order.o
a - amd_l_control.o
a - amd_l_info.o
a - amd_l_valid.o
a - amd_l_preprocess.o
ranlib libamd.a
gcc -L/build/source/lib -Wl,-rpath=/build/source/lib -shared -Wl,-soname -Wl,libamd.so.2 -Wl,--no-undefined -Wl,-rpath,'$ORIGIN' -Wl,-z,origin amd_i_aat.o amd_i_1.o amd_i_2.o amd_i_dump.o amd_i_postorder.o amd_i_defaults.o amd_i_post_tree.o amd_i_order.o amd_i_control.o amd_i_info.o amd_i_valid.o amd_i_preprocess.o amd_l_aat.o amd_l_1.o amd_l_2.o amd_l_dump.o amd_l_postorder.o amd_l_defaults.o amd_l_post_tree.o amd_l_order.o amd_l_control.o amd_l_info.o amd_l_valid.o amd_l_preprocess.o -o /build/source/lib/libamd.so.2.4.6 -lm -lrt -lsuitesparseconfig
( cd /build/source/lib ; ln -sf libamd.so.2.4.6 libamd.so )
( cd /build/source/lib ; ln -sf libamd.so.2.4.6 libamd.so.2 )
cp -f ../Include/amd.h /nix/store/lyv9aigvn1crwlmrr7334902b92izvak-suitesparse-5.13.0-dev/include
cp -f ../Doc/AMD_UserGuide.pdf /build/source/share/doc/suitesparse-5.13.0
cp -f ../README.txt /build/source/share/doc/suitesparse-5.13.0/AMD_README.txt
chmod 755 /build/source/lib/libamd.so.2.4.6
chmod 644 /nix/store/lyv9aigvn1crwlmrr7334902b92izvak-suitesparse-5.13.0-dev/include/amd.h
chmod 644 /build/source/share/doc/suitesparse-5.13.0/AMD_UserGuide.pdf
chmod 644 /build/source/share/doc/suitesparse-5.13.0/AMD_README.txt
make[3]: Leaving directory '/build/source/AMD/Lib'
make[2]: Leaving directory '/build/source/AMD/Lib'
make[1]: Leaving directory '/build/source/AMD'
( cd BTF && make library )
make[1]: Entering directory '/build/source/BTF'
( cd Lib ; make )
make[2]: Entering directory '/build/source/BTF/Lib'
make install INSTALL=/build/source
make[3]: Entering directory '/build/source/BTF/Lib'
gcc    -O3 -fexceptions -fPIC -fopenmp -c -I../Include -I../../SuiteSparse_config ../Source/btf_order.c -o btf_order.o
gcc    -O3 -fexceptions -fPIC -fopenmp -c -I../Include -I../../SuiteSparse_config ../Source/btf_maxtrans.c -o btf_maxtrans.o
gcc    -O3 -fexceptions -fPIC -fopenmp -c -I../Include -I../../SuiteSparse_config ../Source/btf_strongcomp.c -o btf_strongcomp.o
gcc    -O3 -fexceptions -fPIC -fopenmp -c -I../Include -I../../SuiteSparse_config -DDLONG ../Source/btf_order.c -o btf_l_order.o
gcc    -O3 -fexceptions -fPIC -fopenmp -c -I../Include -I../../SuiteSparse_config -DDLONG ../Source/btf_maxtrans.c -o btf_l_maxtrans.o
gcc    -O3 -fexceptions -fPIC -fopenmp -c -I../Include -I../../SuiteSparse_config -DDLONG ../Source/btf_strongcomp.c -o btf_l_strongcomp.o
ar -rv libbtf.a btf_order.o btf_maxtrans.o btf_strongcomp.o btf_l_order.o btf_l_maxtrans.o btf_l_strongcomp.o
ar: creating libbtf.a
a - btf_order.o
a - btf_maxtrans.o
a - btf_strongcomp.o
a - btf_l_order.o
a - btf_l_maxtrans.o
a - btf_l_strongcomp.o
ranlib libbtf.a
gcc -L/build/source/lib -Wl,-rpath=/build/source/lib -shared -Wl,-soname -Wl,libbtf.so.1 -Wl,--no-undefined -Wl,-rpath,'$ORIGIN' -Wl,-z,origin btf_order.o btf_maxtrans.o btf_strongcomp.o btf_l_order.o btf_l_maxtrans.o btf_l_strongcomp.o -o /build/source/lib/libbtf.so.1.2.6 -lm -lrt -lsuitesparseconfig
( cd /build/source/lib ; ln -sf libbtf.so.1.2.6 libbtf.so )
( cd /build/source/lib ; ln -sf libbtf.so.1.2.6 libbtf.so.1 )
cp -f ../Include/btf.h /nix/store/lyv9aigvn1crwlmrr7334902b92izvak-suitesparse-5.13.0-dev/include
cp -f ../README.txt /build/source/share/doc/suitesparse-5.13.0/BTF_README.txt
chmod 755 /build/source/lib/libbtf.so.1.2.6
chmod 644 /nix/store/lyv9aigvn1crwlmrr7334902b92izvak-suitesparse-5.13.0-dev/include/btf.h
chmod 644 /build/source/share/doc/suitesparse-5.13.0/BTF_README.txt
make[3]: Leaving directory '/build/source/BTF/Lib'
make[2]: Leaving directory '/build/source/BTF/Lib'
make[1]: Leaving directory '/build/source/BTF'
( cd CAMD && make library )
make[1]: Entering directory '/build/source/CAMD'
( cd Lib    ; make )
make[2]: Entering directory '/build/source/CAMD/Lib'
make install INSTALL=/build/source
make[3]: Entering directory '/build/source/CAMD/Lib'
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/camd_aat.c -o camd_i_aat.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/camd_1.c -o camd_i_1.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/camd_2.c -o camd_i_2.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/camd_dump.c -o camd_i_dump.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/camd_postorder.c -o camd_i_postorder.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/camd_defaults.c -o camd_i_defaults.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/camd_order.c -o camd_i_order.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/camd_control.c -o camd_i_control.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/camd_info.c -o camd_i_info.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/camd_valid.c -o camd_i_valid.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDINT -c ../Source/camd_preprocess.c -o camd_i_preprocess.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/camd_aat.c -o camd_l_aat.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/camd_1.c -o camd_l_1.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/camd_2.c -o camd_l_2.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/camd_dump.c -o camd_l_dump.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/camd_postorder.c -o camd_l_postorder.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/camd_defaults.c -o camd_l_defaults.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/camd_order.c -o camd_l_order.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/camd_control.c -o camd_l_control.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/camd_info.c -o camd_l_info.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/camd_valid.c -o camd_l_valid.o
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -DDLONG -c ../Source/camd_preprocess.c -o camd_l_preprocess.o
ar -rv libcamd.a camd_i_aat.o camd_i_1.o camd_i_2.o camd_i_dump.o camd_i_postorder.o camd_i_defaults.o camd_i_order.o camd_i_control.o camd_i_info.o camd_i_valid.o camd_i_preprocess.o camd_l_aat.o camd_l_1.o camd_l_2.o camd_l_dump.o camd_l_postorder.o camd_l_defaults.o camd_l_order.o camd_l_control.o camd_l_info.o camd_l_valid.o camd_l_preprocess.o
ar: creating libcamd.a
a - camd_i_aat.o
a - camd_i_1.o
a - camd_i_2.o
a - camd_i_dump.o
a - camd_i_postorder.o
a - camd_i_defaults.o
a - camd_i_order.o
a - camd_i_control.o
a - camd_i_info.o
a - camd_i_valid.o
a - camd_i_preprocess.o
a - camd_l_aat.o
a - camd_l_1.o
a - camd_l_2.o
a - camd_l_dump.o
a - camd_l_postorder.o
a - camd_l_defaults.o
a - camd_l_order.o
a - camd_l_control.o
a - camd_l_info.o
a - camd_l_valid.o
a - camd_l_preprocess.o
ranlib libcamd.a
gcc -L/build/source/lib -Wl,-rpath=/build/source/lib -shared -Wl,-soname -Wl,libcamd.so.2 -Wl,--no-undefined -Wl,-rpath,'$ORIGIN' -Wl,-z,origin camd_i_aat.o camd_i_1.o camd_i_2.o camd_i_dump.o camd_i_postorder.o camd_i_defaults.o camd_i_order.o camd_i_control.o camd_i_info.o camd_i_valid.o camd_i_preprocess.o camd_l_aat.o camd_l_1.o camd_l_2.o camd_l_dump.o camd_l_postorder.o camd_l_defaults.o camd_l_order.o camd_l_control.o camd_l_info.o camd_l_valid.o camd_l_preprocess.o -o /build/source/lib/libcamd.so.2.4.6 -lm -lrt -lsuitesparseconfig
( cd /build/source/lib ; ln -sf libcamd.so.2.4.6 libcamd.so )
( cd /build/source/lib ; ln -sf libcamd.so.2.4.6 libcamd.so.2 )
cp -f ../Include/camd.h /nix/store/lyv9aigvn1crwlmrr7334902b92izvak-suitesparse-5.13.0-dev/include
cp -f ../Doc/CAMD_UserGuide.pdf /build/source/share/doc/suitesparse-5.13.0
cp -f ../README.txt /build/source/share/doc/suitesparse-5.13.0/CAMD_README.txt
chmod 755 /build/source/lib/libcamd.so.2.4.6
chmod 644 /nix/store/lyv9aigvn1crwlmrr7334902b92izvak-suitesparse-5.13.0-dev/include/camd.h
chmod 644 /build/source/share/doc/suitesparse-5.13.0/CAMD_UserGuide.pdf
chmod 644 /build/source/share/doc/suitesparse-5.13.0/CAMD_README.txt
make[3]: Leaving directory '/build/source/CAMD/Lib'
make[2]: Leaving directory '/build/source/CAMD/Lib'
make[1]: Leaving directory '/build/source/CAMD'
( cd CCOLAMD && make library )
make[1]: Entering directory '/build/source/CCOLAMD'
( cd Lib    ; make )
make[2]: Entering directory '/build/source/CCOLAMD/Lib'
make install INSTALL=/build/source
make[3]: Entering directory '/build/source/CCOLAMD/Lib'
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -c ../Source/ccolamd.c
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -c ../Source/ccolamd.c -DDLONG -o ccolamd_l.o
ar -rv  libccolamd.a ccolamd.o ccolamd_l.o
ar: creating libccolamd.a
a - ccolamd.o
a - ccolamd_l.o
ranlib libccolamd.a 
gcc -L/build/source/lib -Wl,-rpath=/build/source/lib -shared -Wl,-soname -Wl,libccolamd.so.2 -Wl,--no-undefined -Wl,-rpath,'$ORIGIN' -Wl,-z,origin ccolamd.o ccolamd_l.o -o /build/source/lib/libccolamd.so.2.9.6 -lm -lrt -lsuitesparseconfig
( cd /build/source/lib ; ln -sf libccolamd.so.2.9.6 libccolamd.so )
( cd /build/source/lib ; ln -sf libccolamd.so.2.9.6 libccolamd.so.2 )
cp -f ../Include/ccolamd.h /nix/store/lyv9aigvn1crwlmrr7334902b92izvak-suitesparse-5.13.0-dev/include
cp -f ../README.txt /build/source/share/doc/suitesparse-5.13.0/CCOLAMD_README.txt
chmod 755 /build/source/lib/libccolamd.so.2.9.6
chmod 644 /nix/store/lyv9aigvn1crwlmrr7334902b92izvak-suitesparse-5.13.0-dev/include/ccolamd.h
chmod 644 /build/source/share/doc/suitesparse-5.13.0/CCOLAMD_README.txt
make[3]: Leaving directory '/build/source/CCOLAMD/Lib'
make[2]: Leaving directory '/build/source/CCOLAMD/Lib'
make[1]: Leaving directory '/build/source/CCOLAMD'
( cd COLAMD && make library )
make[1]: Entering directory '/build/source/COLAMD'
( cd Lib    ; make )
make[2]: Entering directory '/build/source/COLAMD/Lib'
make install INSTALL=/build/source
make[3]: Entering directory '/build/source/COLAMD/Lib'
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -c ../Source/colamd.c
gcc    -O3 -fexceptions -fPIC -fopenmp -I../Include -I../../SuiteSparse_config -c ../Source/colamd.c -DDLONG -o colamd_l.o
ar -rv  libcolamd.a colamd.o colamd_l.o
ar: creating libcolamd.a
a - colamd.o
a - colamd_l.o
ranlib libcolamd.a 
gcc -L/build/source/lib -Wl,-rpath=/build/source/lib -shared -Wl,-soname -Wl,libcolamd.so.2 -Wl,--no-undefined -Wl,-rpath,'$ORIGIN' -Wl,-z,origin colamd.o colamd_l.o -o /build/source/lib/libcolamd.so.2.9.6 -lm -lrt -lsuitesparseconfig
( cd /build/source/lib ; ln -sf libcolamd.so.2.9.6 libcolamd.so )
( cd /build/source/lib ; ln -sf libcolamd.so.2.9.6 libcolamd.so.2 )
cp -f ../Include/colamd.h /nix/store/lyv9aigvn1crwlmrr7334902b92izvak-suitesparse-5.13.0-dev/include
cp -f ../README.txt /build/source/share/doc/suitesparse-5.13.0/COLAMD_README.txt
chmod 755 /build/source/lib/libcolamd.so.2.9.6
chmod 644 /nix/store/lyv9aigvn1crwlmrr7334902b92izvak-suitesparse-5.13.0-dev/include/colamd.h
chmod 644 /build/source/share/doc/suitesparse-5.13.0/COLAMD_README.txt
make[3]: Leaving directory '/build/source/COLAMD/Lib'
make[2]: Leaving directory '/build/source/COLAMD/Lib'
make[1]: Leaving directory '/build/source/COLAMD'
( cd CHOLMOD && make library )
make[1]: Entering directory '/build/source/CHOLMOD'
( cd Lib ; make --jobs=20 )
make[2]: Entering directory '/build/source/CHOLMOD/Lib'
make install INSTALL=/build/source
make[3]: Entering directory '/build/source/CHOLMOD/Lib'
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_aat.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_add.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_band.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_change_factor.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_common.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_complex.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_copy.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_dense.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_error.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_factor.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_memory.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_sparse.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_transpose.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_triplet.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_version.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Check/cholmod_check.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Check/cholmod_read.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Check/cholmod_write.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Cholesky/cholmod_amd.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Cholesky/cholmod_analyze.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Cholesky/cholmod_colamd.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Cholesky/cholmod_etree.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Cholesky/cholmod_factorize.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Cholesky/cholmod_postorder.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Cholesky/cholmod_rcond.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Cholesky/cholmod_resymbol.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Cholesky/cholmod_rowcolcounts.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Cholesky/cholmod_rowfac.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Cholesky/cholmod_solve.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Cholesky/cholmod_spsolve.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../MatrixOps/cholmod_drop.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../MatrixOps/cholmod_horzcat.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../MatrixOps/cholmod_norm.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../MatrixOps/cholmod_scale.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../MatrixOps/cholmod_sdmult.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../MatrixOps/cholmod_ssmult.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../MatrixOps/cholmod_submatrix.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../MatrixOps/cholmod_vertcat.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../MatrixOps/cholmod_symmetry.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Modify/cholmod_rowadd.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Modify/cholmod_rowdel.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Modify/cholmod_updown.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Supernodal/cholmod_super_numeric.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Supernodal/cholmod_super_solve.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Supernodal/cholmod_super_symbolic.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../GPU/cholmod_gpu.c
/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/bin/nvcc -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ -Xcompiler -fPIC -O3 -gencode=arch=compute_53,code=sm_53 -gencode=arch=compute_60,code=compute_60 -c ../GPU/cholmod_gpu_kernels.cu
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Partition/cholmod_ccolamd.c
In file included from /nix/store/pjwsmd212zhpriymn116xfl95zqxcwz5-cuda_cudart-12.4.99-dev/include/cuda_runtime.h:82,
                 from <command-line>:
/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/crt/host_config.h:143:2: error: #error -- unsupported GNU version! gcc versions later than 13 are not supported! The nvcc flag '-allow-unsupported-compiler' can be used to override this version check; however, using an unsupported host compiler may cause compilation failure or incorrect run time execution. Use at your own risk.
  143 | #error -- unsupported GNU version! gcc versions later than 13 are not supported! The nvcc flag '-allow-unsupported-compiler' can be used to override this version check; however, using an unsupported host compiler may cause compilation failure or incorrect run time execution. Use at your own risk.
      |  ^~~~~
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Partition/cholmod_csymamd.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Partition/cholmod_metis.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Partition/cholmod_nesdis.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Partition/cholmod_camd.c
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -DDLONG -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_aat.c -o cholmod_l_aat.o
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -DDLONG -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_add.c -o cholmod_l_add.o
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -DDLONG -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_band.c -o cholmod_l_band.o
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -DDLONG -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_change_factor.c -o cholmod_l_change_factor.o
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -DDLONG -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_common.c -o cholmod_l_common.o
gcc    -O3 -fexceptions -fPIC -fopenmp -DGPU_BLAS  -DDLONG -c -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include -I/build/source/CCOLAMD/Include -I/build/source/CAMD/Include -I../Include -I../../SuiteSparse_config -I/nix/store/vnwm837kbwv0ckf1xbjs6zx9vddcjz8d-cuda_nvcc-12.4.99/include/ ../Core/cholmod_complex.c -o cholmod_l_complex.o
make[3]: *** [Makefile:527: cholmod_gpu_kernels.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/build/source/CHOLMOD/Lib'
make[2]: *** [Makefile:31: library] Error 2
make[2]: Leaving directory '/build/source/CHOLMOD/Lib'
make[1]: *** [Makefile:19: library] Error 2
make[1]: Leaving directory '/build/source/CHOLMOD'
make: *** [Makefile:125: library] Error 2
```

</details>

## Notify maintainers

@ttuegel

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/libraries/science/math/suitesparse/default.nix
+++ b/pkgs/development/libraries/science/math/suitesparse/default.nix
@@ -13,9 +13,13 @@
   enableCuda ? config.cudaSupport,
   cudaPackages,
   openmp ? null,
-}:
+}@inputs:
 
-stdenv.mkDerivation rec {
+let
+  stdenv = throw "Use effectiveStdenv instead";
+  effectiveStdenv = if enableCuda then cudaPackages.backendStdenv else inputs.stdenv;
+in
+effectiveStdenv.mkDerivation rec {
   pname = "suitesparse";
   version = "5.13.0";
 
@@ -35,7 +39,7 @@ stdenv.mkDerivation rec {
   nativeBuildInputs =
     [
     ]
-    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    ++ lib.optionals effectiveStdenv.hostPlatform.isDarwin [
       fixDarwinDylibNames
     ]
     ++ lib.optionals enableCuda [
@@ -53,7 +57,7 @@ stdenv.mkDerivation rec {
       gmp
       mpfr
     ]
-    ++ lib.optionals stdenv.cc.isClang [
+    ++ lib.optionals effectiveStdenv.cc.isClang [
       openmp
     ]
     ++ lib.optionals enableCuda [
@@ -82,14 +86,14 @@ stdenv.mkDerivation rec {
       "CUDART_LIB=${lib.getLib cudaPackages.cuda_cudart}/lib/libcudart.so"
       "CUBLAS_LIB=${lib.getLib cudaPackages.libcublas}/lib/libcublas.so"
     ]
-    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    ++ lib.optionals effectiveStdenv.hostPlatform.isDarwin [
       # Unless these are set, the build will attempt to use `Accelerate` on darwin, see:
       # https://github.com/DrTimothyAldenDavis/SuiteSparse/blob/v5.13.0/SuiteSparse_config/SuiteSparse_config.mk#L368
       "BLAS=-lblas"
       "LAPACK=-llapack"
     ];
 
-  env = lib.optionalAttrs stdenv.hostPlatform.isDarwin {
+  env = lib.optionalAttrs effectiveStdenv.hostPlatform.isDarwin {
     # Ensure that there is enough space for the `fixDarwinDylibNames` hook to
     # update the install names of the output dylibs.
     NIX_LDFLAGS = "-headerpad_max_install_names";

